pkg/printers/humanreadable.go

```diff
@@ -201,8 +204,51 @@ func (h *HumanReadablePrinter) PrintObj(obj runtime.Object, output io.Writer) er
 	}
 
 	if _, err := meta.Accessor(obj); err == nil {
+		// we don't recognize this type, but we can still attempt to print some reasonable information about.
+		unstructured, ok := obj.(runtime.Unstructured)
+		if !ok {
+			return fmt.Errorf("error: unknown type %#v", obj)
+		}
+
+		content := unstructured.UnstructuredContent()
+
+		// we'll elect a few more fields to print depending on how much columns are already taken
+		maxDiscoveredFieldsToPrint := 3
+		maxDiscoveredFieldsToPrint = maxDiscoveredFieldsToPrint - len(h.options.ColumnLabels)
+		if h.options.WithNamespace { // where's my ternary
+			maxDiscoveredFieldsToPrint--
+		}
+		if h.options.ShowLabels {
+			maxDiscoveredFieldsToPrint--
+		}
+		if maxDiscoveredFieldsToPrint < 0 {
+			maxDiscoveredFieldsToPrint = 0
+		}
+
+		var discoveredFieldNames []string                    // we want it predictable so this will be used to sort
+		ignoreIfDiscovered := []string{"kind", "apiVersion"} // these are already covered
+		for field, value := range content {
+			if slice.ContainsString(ignoreIfDiscovered, field, nil) {
```

Comment:
Yes, here in get the order is always the same so it will always discover the same fields.